src/pages/workspace/distanceRates/PolicyDistanceRateEditPage.tsx

```diff
@@ -0,0 +1,201 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useState} from 'react';
+import {Keyboard, View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import ConfirmModal from '@components/ConfirmModal';
+import type {FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import * as Expensicons from '@components/Icon/Expensicons';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Switch from '@components/Switch';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as CurrencyUtils from '@libs/CurrencyUtils';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import FeatureEnabledAccessOrNotFoundWrapper from '@pages/workspace/FeatureEnabledAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+import type {Rate} from '@src/types/onyx/Policy';
+import RateModal from './RateModal';
+
+type PolicyDistanceRateEditPageOnyxProps = {
+    /** Policy details */
+    policy: OnyxEntry<OnyxTypes.Policy>;
+};
+
+type PolicyDistanceRateEditPageProps = PolicyDistanceRateEditPageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.DISTANCE_RATE_EDIT>;
+
+function PolicyDistanceRateEditPage({policy, route}: PolicyDistanceRateEditPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {windowWidth} = useWindowDimensions();
+    const [isWarningModalVisible, setIsWarningModalVisible] = useState(false);
+    const [isRateModalVisible, setIsRateModalVisible] = useState(false);
+    const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
+
+    const policyID = route.params.policyID;
+    const rateID = route.params.rateID;
+    const customUnits = policy?.customUnits ?? {};
+    const customUnit = customUnits[Object.keys(customUnits)[0]];
+    const rate = customUnit.rates[rateID];
+    const currency = rate.currency ?? CONST.CURRENCY.USD;
+    const canDeleteRate = Object.keys(customUnit.rates).length > 1;
```

Comment:
```suggestion
    const canDeleteRate = Object.values(customUnit.rates).filter(rate => rate.enabled).length > 1;
```

Based on the modal message we need at least one enabled rate, so if we have two distance rates and one is enabled i think we shouldn't be able to delete the enabled one because it will leave us with the disabled one.